modules/react/common/lib/theming/createCanvasTheme.ts

```diff
@@ -103,7 +101,7 @@ function calculateCanvasTheme(partialTheme: PartialCanvasTheme): CanvasTheme {
     direction: direction === ContentDirection.RTL ? direction : ContentDirection.LTR,
   };
 
-  return merge({}, defaultCanvasTheme, mergeable, extraFields) as CanvasTheme;
+  return getObjectProxy({...mergeable, ...extraFields}, defaultCanvasTheme);
 }
 
-export const createCanvasTheme = memoize(calculateCanvasTheme, (...args) => JSON.stringify(args));
+export const createCanvasTheme = calculateCanvasTheme;
```

Comment:
Going based on @NicholasBoll comment that we lose a lot of our memoizing benefits because of JSON.stringify.